dts/client.py

```diff
@@ -154,42 +177,51 @@ def search(self,
     def transfer(self,
                  file_ids = None,
                  source = None,
-                 destination = None):
+                 destination = None,
+                 timeout = None):
         """
 `client.transfer(file_ids = None,
                  source = None,
-                 destination = None) -> UUID
+                 destination = None,
+                 timeout = None) -> UUID
 
 * Submits a request to transfer files from a source to a destination database. the
   files in the source database are identified by a list of string file_ids.
 """
         if not self.uri:
             raise RuntimeError('dts.Client: not connected.')
-        if type(source) != str:
+        if not isinstance(source, str):
             raise TypeError('transfer: source database name must be a string.')
-        if type(destination) != str:
+        if not isinstance(destination, str):
             raise TypeError('transfer: destination database name must be a string.')
-        if type(file_ids) != list:
-            raise TypeError('batch: sequences must be a list of string file IDs.')
+        if not isinstance(file_ids, list):
+            raise TypeError('transfer: file_ids must be a list of string file IDs.')
+        if timeout and not isinstance(timeout, int) and not isinstance(timeout, float):
+            raise TypeError('transfer: timeout must be a number of seconds.')
         try:
-            response = requests.post(f'{self.uri}/transfers',
-                                     data={
-                                         source:      source,
-                                         destination: destination,
-                                         file_ids:    file_ids,
-                                     })
+            response = requests.post(url=f'{self.uri}/transfers',
+                                     json={
+                                         'source':      source,
+                                         'destination': destination,
+                                         'file_ids':    file_ids,
+                                     },
+                                     auth=self.auth,
+                                     timeout=timeout)
             response.raise_for_status()
         except HTTPError as http_err:
             logger.error(f'HTTP error occurred: {http_err}')
             return None
+        except requests.exceptions.HTTPError as err:
+            logger.error(f'HTTP error occurred: {err}')
+            return None
         except Exception as err:
             logger.error(f'Other error occurred: {err}')
             return None
         else:
             return uuid.UUID(response.json()["id"])
```

Comment:
all the `except` clauses return something (well, None), so no need for the `else` here